tsl/test/expected/compression.out

```diff
@@ -926,6 +926,37 @@ NOTICE:  adding not-null constraint to column "time"
 (1 row)
 
 INSERT INTO metrics SELECT generate_series('2000-01-01'::timestamptz,'2000-01-10','1m'),1,0.25,0.75;
+CREATE MATERIALIZED VIEW cagg_expr WITH (timescaledb.continuous)
+AS
+SELECT
+  time_bucket('1d', time) AS time,
+  'Const'::text AS Const,
+  4.3::numeric AS "numeric",
+  first(metrics,time),
+  CASE WHEN true THEN 'foo' ELSE 'bar' END,
+  COALESCE(NULL,'coalesce'),
+  avg(v1) + avg(v2) AS avg1,
+  avg(v1+v2) AS avg2,
+  count(*) AS cnt
+FROM metrics
+WHERE ctid < '(1,1)'::tid GROUP BY 1 WITH NO DATA;
+CALL refresh_continuous_aggregate('cagg_expr', NULL, NULL);
+SET timescaledb.enable_chunk_append to off;
+SET enable_indexscan to off;
+-- github issue 4872
+-- If subplan of ConstraintAwareAppend is TidRangeScan, then SELECT on
+-- hypertable fails with error "invalid child of chunk append: Node (26)"
+SELECT * FROM cagg_expr ORDER BY time LIMIT 5;
+             time             | const | numeric |                    first                     | case | coalesce | avg1 | avg2 | cnt 
+------------------------------+-------+---------+----------------------------------------------+------+----------+------+------+-----
+ Fri Dec 31 16:00:00 1999 PST | Const |     4.3 | ("Sat Jan 01 00:00:00 2000 PST",1,0.25,0.75) | foo  | coalesce |    1 |    1 | 136
+ Wed Jan 05 16:00:00 2000 PST | Const |     4.3 | ("Wed Jan 05 16:00:00 2000 PST",1,0.25,0.75) | foo  | coalesce |    1 |    1 | 136
+(2 rows)
+
+RESET timescaledb.enable_chunk_append;
+RESET enable_indexscan;
+DROP MATERIALIZED VIEW cagg_expr;
+NOTICE:  drop cascades to table _timescaledb_internal._hyper_14_35_chunk
```

Comment:
You might want to consider adding new tests last to avoid cascading result set changes because of hypertable id changes.

Reply:
Fixed.